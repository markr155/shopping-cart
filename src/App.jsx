import NavBar from './Components/Navbar';
import { Outlet } from 'react-router-dom';
import { useState } from 'react';

const App = () => {
  const [cart, setCart] = useState([]);
  const handleAddToCart = (name, image, price, description, quantity) => {
		const newItem = {
			key: name,
			name,
			image,
			price,
			description,
			quantity
		}
		setCart(prevCart => [...prevCart, newItem]);
	};

  return (
    <>
      <NavBar cart={cart} />
      <Outlet context={{
				cart: [cart, setCart],
				handleAddToCart: handleAddToCart
				}}
				 />
    </>
  );
};

export default App;
