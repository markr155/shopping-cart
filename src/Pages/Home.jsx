import { Link } from 'react-router-dom';
export default function Home() {
	return (
		<>
			<h1>Home</h1>
			<img alt='shop image'></img>
			<button>
				<Link to='store'>Shop Now</Link>
			</button>
		</>
	);
}
