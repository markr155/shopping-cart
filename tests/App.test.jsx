import App from '../src/App';
import NavBar from '../src/Components/Navbar';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Home from '../src/Pages/Home';
import { it } from 'vitest';

describe('Default page loads', () => {
	it('NavBar renders', () => {
		const { container } = render(<NavBar cart={[]} />, {
			wrapper: BrowserRouter,
		});
		expect(container).toMatchSnapshot();
	});

	it('Home renders', () => {
		const { container } = render(<Home />, { wrapper: BrowserRouter });
		expect(container).toMatchSnapshot();
	});
});
