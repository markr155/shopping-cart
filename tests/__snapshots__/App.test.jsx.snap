// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Default page loads > Home renders 1`] = `
<div>
  <h1>
    Home
  </h1>
  <img
    alt="shop image"
  />
  <button>
    <a
      href="/store"
    >
      Shop Now
    </a>
  </button>
</div>
`;

exports[`Default page loads > NavBar renders 1`] = `
<div>
  <div>
    <h1>
      Store Name
    </h1>
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/store"
        >
          Store
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/cart"
        >
          Cart
        </a>
      </li>
    </ul>
  </div>
</div>
`;
